x-pack/platform/plugins/shared/stack_connectors/server/connector_types/slack_api/service.test.ts

```diff
@@ -388,4 +393,315 @@ describe('Slack API service', () => {
+
+        await service[method]({ [key]: [value], text });
+
+        expect(requestMock).toHaveBeenCalledTimes(1);
+
+        expect(requestMock).toHaveBeenCalledWith(
+          expect.objectContaining({
+            data: expect.objectContaining({ channel: value }),
+          })
+        );
+      });
+
+      it('should post a message if allowedChannels is an empty array and use channelsIds', async () => {
+        service = createExternalServiceMock({ config: {} });
+
+        await service[method]({ channelIds: ['channel-id-1'], text });
+
+        expect(requestMock).toHaveBeenCalledTimes(1);
+
+        expect(requestMock).toHaveBeenCalledWith(
+          expect.objectContaining({
+            data: expect.objectContaining({ channel: 'channel-id-1' }),
+          })
+        );
+      });
+
+      it('should post a message if allowedChannels is an empty array and use channels', async () => {
+        service = createExternalServiceMock({ config: {} });
+
+        await service[method]({ channels: ['my-channel'], text });
+
+        expect(requestMock).toHaveBeenCalledTimes(1);
+
+        expect(requestMock).toHaveBeenCalledWith(
+          expect.objectContaining({
+            data: expect.objectContaining({ channel: 'my-channel' }),
+          })
+        );
+      });
+
+      it('should throw an error if channelNames is not in allowedChannels', async () => {
+        service = createExternalServiceMock({
+          config: { allowedChannels: [{ name: 'channel-name' }] },
+        });
+
+        expect(await service[method]({ channelNames: ['not-in-list'], text })).toEqual(
+          errorNoAllowedChannelsRes
+        );
+      });
+
+      it('should throw an error if channelIds is not in allowedChannels', async () => {
+        service = createExternalServiceMock({
+          config: { allowedChannels: [{ id: 'channel-1-id', name: 'channel-name' }] },
+        });
+
+        expect(await service[method]({ channelIds: ['not-in-list'], text })).toEqual(
+          errorNoAllowedChannelsRes
+        );
+      });
+
+      it('should validate against allowedChannels.name for channelNames', async () => {
+        service = createExternalServiceMock({
+          config: { allowedChannels: [{ id: 'channel-1-id', name: 'my-channel-name' }] },
+        });
+
+        expect(await service[method]({ channelNames: ['channel-1-id'], text })).toEqual(
+          errorNoAllowedChannelsRes
+        );
+      });
+
+      it('should validate against allowedChannels.id for channelIds', async () => {
+        service = createExternalServiceMock({
+          config: { allowedChannels: [{ id: 'channel-1-id', name: 'my-channel-name' }] },
+        });
+
+        expect(await service[method]({ channelIds: ['my-channel-name'], text })).toEqual(
+          errorNoAllowedChannelsRes
+        );
+      });
+
+      it('should not throw if allowedChannels.name is set and channelIds is used', async () => {
```

Comment:
Why?

And what about `should not throw if allowedChannels.ids is set and channelNames is used`?